source/extensions/filters/http/ext_proc/processor_state.cc

```diff
@@ -603,6 +603,36 @@ bool ProcessorState::handleDuplexStreamedBodyResponse(const CommonResponse& comm
   return end_of_stream;
 }
 
+bool ProcessorState::isLastResponseAfterHeaderResp() {
+  if (hasNoBody()) {
+    return true;
+  }
+  if (bodyMode() == ProcessingMode::NONE && !sendTrailers()) {
+    return true;
+  }
+  if (bodyMode() == ProcessingMode::NONE && sendTrailers()) {
+    if (completeBodyAvailable() && (responseTrailers() == nullptr)) {
+      return true;
+    }
+  }
+  if (bodyMode() != ProcessingMode::NONE && !sendTrailers()) {
```

Comment:
if body is already waiting to be sent(AND trailers received), would this be an issue?

Reply:
If there is body waiting to be send, and body mode is not NONE, then Envoy will send the body when processing the header response, and the callback state will not be IDLE.

Reply:
done